### PR TITLE
Add useWallet and Faucet to casper-open-source-software doc

### PR DESCRIPTION
**Pull request by hoailinh1210**
### What does this PR fix/introduce?
This PR adds two new sections to the `casper-open-source-software.md` document, providing information about two CasperDash resources:

- [UseWallet](https://usewallet.casperdash.io): useWallet is a collection of React Hooks containing everything you need to start working with Casper Network.
- [Faucet](https://faucet.casperdash.io): A faucet is a Casper tool that drips testnet tokens to anyone that requests them for free.

### Additional context
No additional context is required for this simple documentation update.

### Checklist
(Delete any that aren't relevant)

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casper.network/resources/contribute-to-docs/).

### CODE REVIEW

**Review by Jiuhong-casperlabs — APPROVED** [review]
It looks good to me.

File: source/docs/casper/resources/casper-open-source-software.md [hunk 1]
```diff
@@ -62,3 +62,5 @@ Name | Description | Author | Language | License | Last Update Date | Type
 [ServicesDAO](https://github.com/EKON-YAZILIM/ServicesDAO) | The ServicesDAO portal provides a platform for the service DAOs which provide services in a decentralized manner through a modular implementation of MVPR. | Ekon Yazilim | HTML-CSS-C# | MIT license | 2022-09-13 | DAO
 [Uniswap DemoApp](https://github.com/Rengo-Labs/CasperLabs-Uniswap-DemoApp) | Uniswap UI + Contracts | Rengo Labs | JavaScript | GPL-3.0 license | 2022-09-15 | DeFi
 [Verified Impact NFT](https://github.com/AdelElMessiry/Verified-Impact-NFTs) | An NFT platform dedicated to impact causes with verification of the beneficiaries | AlphaFin | JavaScript-Rust | MIT license | 2022-07-08 | NFT
+[UseWallet](https://usewallet.casperdash.io) | useWallet is a collection of React Hooks containing everything you need to start working with Casper Network | CasperDash | Typescript-React | MIT license | 2023-09-19 | dApp library
```
Comment (ipopescu) [inline source/docs/casper/resources/casper-open-source-software.md:65]:
```suggestion
[UseWallet](https://usewallet.casperdash.io) | useWallet is a collection of React Hooks containing everything you need to start working with a Casper network | CasperDash | Typescript-React | MIT license | 2023-09-19 | dApp library
```

File: source/docs/casper/resources/casper-open-source-software.md [hunk 2]
```diff
@@ -62,3 +62,5 @@ Name | Description | Author | Language | License | Last Update Date | Type
...
+[Faucet](https://faucet.casperdash.io) | A faucet is a casper tool that drips testnet tokens to anyone that requests them for free | CasperDash | Typescript-React | MIT license | 2023-09-19 | Tools
```
Comment (ipopescu) [inline source/docs/casper/resources/casper-open-source-software.md:66]:
```suggestion
[Testnet Faucet](https://faucet.casperdash.io) | A faucet is a Casper tool that distributes Testnet tokens to anyone requesting them for free | CasperDash | Typescript-React | MIT license | 2023-09-19 | Tools
```

**Review by ipopescu — APPROVED** [review]
Approved after some minor updates

**Pull request merged**